modules/tinymce/src/plugins/quickbars/main/ts/api/Commands.ts

```diff
@@ -0,0 +1,25 @@
+import Editor from 'tinymce/core/api/Editor';
+
+import * as Actions from '../insert/Actions';
+import * as Conversions from '../insert/Conversions';
+import * as Picker from '../insert/Picker';
+
+const register = (editor: Editor): void => {
+  if (!editor.queryCommandSupported('QuickbarInsertImage')) {
+    editor.addCommand('QuickbarInsertImage', () => {
```

Comment:
I suggest delaying registration of this command to `preinit` - otherwise we have another race condition based on plugin load order.

- [ ] delay to right before the editor finishes loading

Reply:
It might not help if another plugin wants to override the command in `preinit`, right?

Reply:
There will always be _some_ sort of race condition, but our examples all register event handlers and commands when the plugin loads - which is long before `preinit`.

Reply:
wait, I added [a setup implementation to the test case](https://github.com/tinymce/tinymce/pull/9974/files#diff-693762719f6da463204bea6c874fd4becd9ebe19fabea4321128807de33cd2b8R18-R20) and I did it in this way to be sure that if another plugin implement `QuickbarInsertImage` the new implementation always wins.

at the moment the behavior is:
`newImplementation` -then> `quickbars` loads => the `if` check is false
`quickbars` loads -then> `newImplementation` => new implementation will overwrite the `quickbars`' one

the only way to have a race condition that I see is if that `addCommand` is async, Am I missing something?

Reply:
Oh you did `preinit` in the new plugin implementation to avoid the race condition. I'd reverse that and have the `preinit` here instead.

Reply:
moved inside the `PreInit`